Snowflake.Data/Core/Session/SFSessionProperty.cs

```diff
@@ -292,6 +265,49 @@ internal static SFSessionProperties parseConnectionString(String connectionStrin
             return properties;
         }
 
+        private static void ProcessPropertySpecialCases(string connectionString, SFSessionProperties properties)

```

Comment:
I like that you thought of extracting some block of code into a separate method.
However this method does a tricky thing - it updates its parameter - properties values.
Normally we do not expect that a method updates its parameters - in such case a method usually returns the modified object.
So it would be good to give it a more descriptive name at least. What would say about `UpdatePropertiesForSpecialCases` ?

Reply:
I renamed the method with the suggested name. Thank you